internal/util/openapi/openapi_test.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openapi
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/go-openapi/spec"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/rest/fake"
+	cmdtesting "k8s.io/kubectl/pkg/cmd/testing"
+	"sigs.k8s.io/kustomize/kyaml/openapi"
+)
+
+func TestSomething(t *testing.T) {
+	testCases := []struct {
+		name         string
+		schemaSource string
+		schemaPath   string
+		response     *http.Response
+		refString    string
+		expectError  bool
+	}{
+		{
+			name:         "no schemaSource provided with fallback to builtin",
+			schemaSource: "",
+			response: &http.Response{StatusCode: http.StatusNotFound,
+				Header: cmdtesting.DefaultHeader(), Body: cmdtesting.StringBody("")},
+			refString:   "#/definitions/io.k8s.api.core.v1.PodSpec",
+			expectError: false,
+		},
+		{
+			name:         "no schemaSource provided with cluster available",
+			schemaSource: "",
+			response: &http.Response{StatusCode: http.StatusOK,
+				Header: cmdtesting.DefaultHeader(), Body: getSchema(t, "clusterschema.json")},
+			refString:   "#/definitions/io.k8s.clusterSchema",
+			expectError: false,
+		},
+		{
+			name:         "schemaSource cluster with successful fetch",
+			schemaSource: "cluster",
+			response: &http.Response{StatusCode: http.StatusOK,
+				Header: cmdtesting.DefaultHeader(), Body: getSchema(t, "clusterschema.json")},
+			refString:   "#/definitions/io.k8s.clusterSchema",
+			expectError: false,
+		},
+		{
+			name:         "schemaSource cluster with failed fetch",
+			schemaSource: "cluster",
+			response: &http.Response{StatusCode: http.StatusNotFound,
+				Header: cmdtesting.DefaultHeader(), Body: cmdtesting.StringBody("")},
+			expectError: true,
+		},
+		{
+			name:         "schemaSource file with valid path",
```

Comment:
We should have a negative test where looking for #/definitions/io.k8s.api.core.v1.PodSpec should throw error, to make sure that we are not using builtin schema.

Reply:
Fixed